contracts/modules/lp_shared.vy

```diff
@@ -139,3 +144,72 @@ def _fetch_nfpm(_factory: address) -> address:
     return abi_decode(response, address)
 
   return empty(address)
+
+@internal
+@view
+def _voter_gauge_to_incentive(_gauge: address) -> address:
+  """
+  @notice Handles root/leaf voter call to gaugeToBribe/gaugeToIncentive
+  @return Incentive contract address
+  """
+  if chain.id in ROOT_CHAIN_IDS:
+    return staticcall self.voter.gaugeToBribe(_gauge)
+
+  return staticcall self.voter.gaugeToIncentive(_gauge)
+
+@internal
+@view
+def _root_lp_address(
+  _factory: address,
+  _token0: address,
+  _token1: address,
+  _type: int24
+) -> address:
+  """
+  @notice Calculates the corresponding root (placeholder) pool address
+  @param _factory The factory address
+  @param _token0 The pool token0
+  @param _token1 The pool token1
+  @param _type The pool type
+  @return address
+  """
+  init_hash: bytes32 = staticcall self.registry.initHashToPoolFactory(_factory)
+
+  if init_hash == empty(bytes32):
+    return empty(address)
+
+
+  salt: bytes32 = empty(bytes32)
+
+  if _type < 1:
```

Comment:
You could rewrite the `if-else` block like this:

```vyper
common: Bytes[72] = concat(convert(chain.id, bytes32), convert(_token0, bytes20), convert(_token1, bytes20))
salt = keccak256(concat(common, 0x00)) if (_type < 1) else keccak256(concat(common, convert(_type, bytes3)))
```

Probably the recommended approach consumes slightly more gas, but is definitely more readable imo.